app/launcher/Main.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Main where
+
+import CLI
+    ( Decodable (decode), Encodable (encode), Network, Port (..), getArg )
+import Control.Concurrent
+    ( forkIO )
+import Control.Monad
+    ( forM, forM_, when )
+import GHC.IO.Handle
+    ( Handle )
+import Prelude
+import System.Console.Docopt
+    ( Docopt, docopt, exitWithUsage, isPresent, longOption, parseArgsOrExit )
+import System.Environment
+    ( getArgs )
+import System.Exit
+    ( exitWith )
+import System.Process
+    ( CreateProcess
+    , ProcessHandle
+    , cleanupProcess
+    , createProcess
+    , interruptProcessGroupOf
+    , proc
+    , waitForProcess
+    )
+
+
+-- | Command-Line Interface specification. See http://docopt.org/
+cli :: Docopt
+cli = [docopt|
+cardano-wallet-launcher
+
+Start the cardano wallet along with its API and underlying node.
+
+Requires cardano-http-bridge. To install, follow instructions at
+https://github.com/input-output-hk/cardano-http-bridge, and run
+  cargo install --path .
+in the directory.
+
+
+Usage:
+  cardano-wallet-launcher [options]
+  cardano-wallet-launcher --help
+
+Options:
+  --wallet-server-port <PORT>  port used for serving the wallet API [default: 8090]
+  --node-port <PORT>           port used for node-wallet communication [default: 8080]
+  --network <NETWORK>          mainnet or testnet [default: mainnet]
+|]
+
+
+type Process = (Maybe Handle, Maybe Handle, Maybe Handle, ProcessHandle)
```

Comment:
I did this for [`cleanupProcess`](http://hackage.haskell.org/package/process-1.6.5.0/docs/System-Process.html#v:cleanupProcess). Not sure if it is needed.